frb_codegen/src/generator/dart/ty_opaque.rs

```diff
@@ -0,0 +1,62 @@
+use crate::generator::dart::ty::*;
+use crate::ir::*;
+use crate::target::Acc;
+use crate::type_dart_generator_struct;
+
+type_dart_generator_struct!(TypeOpaqueGenerator, IrTypeOpaque);
+
+impl TypeDartGeneratorTrait for TypeOpaqueGenerator<'_> {
+    fn api2wire_body(&self) -> Acc<Option<String>> {
+        Acc {
+            io: Some(format!(
+                "final ptr = inner.new_{0}();
+                _api_fill_to_wire_{0}(raw, ptr);
+                return ptr;",
+                self.ir.safe_ident(),
+            )),
+            wasm: Some("return raw.tryShare();".to_owned()),
```

Comment:
in theory, we can use ffi.Pointer directly on the native platform.
then the implementation will be the same, but in the io.rs instead of understandable types there will be *mut c_void

```
#[no_mangle]
pub extern "C" fn wire_run_opaque(port_: i64, opaque: *mut c_void)  { ... }
```
instead
```
|#[no_mangle]
pub extern "C" fn wire_run_opaque(port_: i64, opaque: wire_HideData) { ... }
```

also IIRC there will be a conflict with
```
impl<T, S> Wire2Api<Option<T>> for *mut S
where
    *mut S: Wire2Api<T>,
{
    fn wire2api(self) -> Option<T> {
        (!self.is_null()).then(|| self.wire2api())
    }
}
```

Reply:
I see. strong typing is always important.